distributed/nanny.py

```diff
@@ -418,18 +346,12 @@ def run():
 
 
 def isalive(proc):
-    if proc is None:
-        return False
-    elif isinstance(proc, subprocess.Popen):
-        return proc.poll() is None
-    else:
-        return proc.is_alive()
+    return proc is not None and proc.is_alive()
 
 
 def join(proc, timeout):
-    if proc is None or isinstance(proc, subprocess.Popen):
-        return
-    proc.join(timeout)
+    if proc is not None:
+        proc.join(timeout)
```

Comment:
They still handle the case where proc is `None`. I don't know if that's a good enough motivation.